examples/chip-tool/commands/discover/DiscoverCommissionablesCommand.h

```diff
@@ -121,10 +125,16 @@ class DiscoverCommissionableByCommissioningModeCommand : public DiscoverCommissi
 public:
     DiscoverCommissionableByCommissioningModeCommand(CredentialIssuerCommands * credsIssuerConfig) :
         DiscoverCommissionablesCommandBase("find-commissionable-by-commissioning-mode", credsIssuerConfig)
-    {}
+    {
+        AddArgument("commissioning-mode", chip::to_underlying(chip::Dnssd::CommissioningMode::kEnabledBasic),
+                    chip::to_underlying(chip::Dnssd::CommissioningMode::kEnabledJointFabric), &mCommissioningMode);
```

Comment:
This adds a new required argument, which will break everyone using this command right now.  That's not OK.

1. This argument should be optional.
2. If that's done, it having a value can then be used as a signal that we are supposed to be in `kCommissioningMode`, so we don't need `mFilter` at all and most of the changes in this PR can go away, right?

Reply:
Agreed - updated to make commissioning mode optional.